### ISSUE TITLE
[BUG] Breaking error when the specified code block language is misspelled/doesn't exist

### ISSUE THREAD

**Issue by enzocomics** [mdx-editor/editor#423]
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] I have read the documentation and cannot find an answer.

**Describe the bug**
When typing a code block in the editor, if the user misspells the language, or references a language that doesn't exist, an application error occurs, necessitating a complete page refresh: 
```
Application error: a client-side exception has occurred (see the browser console for more information).
```
Console yields this error: `Error: No CodeBlockEditor registered for language=html meta=`

**Reproduction**
I discovered it on my dev site, but the issue is reproducible on the demo site https://mdxeditor.dev/editor/demo

**To Reproduce**
1. In the editor, on a new line, type ` ```html`, followed by spacebar

**Expected behavior**
- Specifying a code block language that isn't defined in `codeBlockLanguages` shouldn't throw an app-breaking error
- Maybe it shouldn't even attempt to create a code block if the language is not specified

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Browser: Firefox 125.0.1, Chrome 124.0.6367.61

**Additional context**
- The issue was referenced here, but it wasn't the original topic of the thread:
   - https://github.com/mdx-editor/editor/issues/290#issuecomment-1977964795 and 
   - https://github.com/mdx-editor/editor/issues/290#issuecomment-1984723089
- Similar issue in this thread: https://github.com/mdx-editor/editor/issues/89 
- Semi-related discussion: https://github.com/mdx-editor/editor/discussions/196

Thank you for your help!

**Comment by petyosi** [comment]
You raise a fair point here. I think I can just make it use the default or plain text if the language is unrecognized. What do you think?

**Comment by ajboni** [comment]
> You raise a fair point here. I think I can just make it use the default or plain text if the language is unrecognized. What do you think?

I think that would be the expected behavior. As a reference this is how obsidian deals with these cases:

![image](https://github.com/mdx-editor/editor/assets/12578313/68cba881-84fe-4002-a1c5-9cf209fc419e)

**Comment by Melvynx** [comment]
Yes, I totally agree. I encountered many issues using this, so I added literally every language 😂 Just the syntax color is not perfect.

**Comment by enzocomics** [comment]
Yes, I agree as well. It should definitely revert to the default language (or plain text, if the language is unspecified). Thanks for looking into it!

**Comment by petyosi** [comment]
Everyone, just published [3.0.0-alpha.1](https://github.com/mdx-editor/editor/compare/v2.20.6...v3.0.0-alpha.1). It replaces the previously used sandpack wrapper with vanilla a codemirror editor. This is a breaking change, because the codemirror plugin will no longer accept a sandpack theme. 

The release does a few other things:

1. Fixes the issue here, by defaulting to the `defaultLanguage`, or to none
2. Automatically loads the necessary language extension (can be disabled)
3. Lets you pass additional code mirror extensions

Please test this and let me know if you encounter any issues.

**Comment by enzocomics** [comment]
Confirming that the update resolved my issue. Thank you so much @petyosi for your awesome work!
